src/components/Vega/EntropyPlot.js

```diff
@@ -314,11 +277,73 @@ const EntropyPlot = observer(({ width }) => {
     }
   };
 
+  const checkIfPrimersOverlap = (primer1, primer2) => {
+    return (
+      (primer1.Start < primer2.End && primer1.End > primer2.End) ||
+      (primer1.Start < primer2.Start && primer1.End > primer2.Start)
+    );
+  };
+
+  const getPrimers = () => {
+    const selectedPrimers = configStore.selectedPrimers;
+
+    if (selectedPrimers.length) {
+      selectedPrimers[0].row = 0;
+      for (let i = 0; i < selectedPrimers.length; i++) {
+        let overlaps = true;
+        let curRow = 0;
+        const primerToPlace = selectedPrimers[i];
+
+        while (overlaps) {
+          const primersInRow = selectedPrimers.filter(
+            (primer) => primer.row && primer.row === curRow
+          );
+
+          if (primersInRow.length) {
+            for (const primer of primersInRow) {
+              overlaps = checkIfPrimersOverlap(primer, primerToPlace);
+              if (!overlaps) break;
+            }
+          } else {
+            overlaps = false;
+          }
+
+          if (overlaps) curRow += 1;
+        }
+
+        primerToPlace.row = curRow;
+        primerToPlace.ranges = [[primerToPlace.Start, primerToPlace.End]];
+        primerToPlace.name = primerToPlace.Name;
+        selectedPrimers[i] = primerToPlace;
+      }
+      return selectedPrimers;
+    } else {
+      const nullDomain = [
+        {
+          Institution: 'None',
+          Name: 'No Primers Selected',
+          ranges: [[0, 30000]],
+          row: 0,
+          Start: 0,
+          End: 30000,
+        },
+      ];
+      configStore.selectedPrimers = nullDomain;
+      return nullDomain;
```

Comment:
Do we have to show this null domain? Doesn't the plot not display if no mutations are returned?

Reply:
There's a weird behavior that happens if an empty array is returned (attached a screenshot) returning the nullDomain makes everything work as intended and I like it for UX reasons
<img width="1355" alt="Screen Shot 2021-12-07 at 3 47 47 PM" src="https://user-images.githubusercontent.com/64983484/145103797-499b92f7-0ea8-4610-8a54-2a8cdd09fbdd.png">

Reply:
More weird behavior: highlighting the domain plot after returning an empty array
<img width="1357" alt="Screen Shot 2021-12-07 at 3 49 34 PM" src="https://user-images.githubusercontent.com/64983484/145104008-9e5620b6-6ed1-4366-8642-1b5f33e87dec.png">

Reply:
When does the 2nd case occur?

I just opened an issue that will not allow users to not select any primers when in primer mode, because I don't think we should accommodate this edge case. But I'm ok with leaving code in

Reply:
Both screenshots are from the case where someone selects primer mode but does not select any primers and the nullDomain code is not included. The 2nd screenshot is what happens when you click and drag on the domain plot in this case.

Including the nullDomain code results in the plot working as intended:
<img width="1340" alt="Screen Shot 2021-12-07 at 3 57 44 PM" src="https://user-images.githubusercontent.com/64983484/145105068-c4ea0ce2-cada-4bbc-937f-490cc53abb92.png">

And yeah, we should definitely prevent users from selecting no primers in primer mode, it defeats the purpose.